test/dialects/postgres_test.py

```diff
@@ -30,6 +30,9 @@
         ("SelectClauseElementSegment", "c is not null as c_notnull"),
         ("SelectClauseElementSegment", "c isnull as c_isnull"),
         ("SelectClauseElementSegment", "c notnull as c_notnull"),
+        ("ArrayAccessorSegment", "[2:10]"),
+        ("ArrayAccessorSegment", "[:10]"),
+        ("ArrayAccessorSegment", "[2:]"),
```

Comment:
Can you add a test here for the simple case of `[2]`, which I worry wouldn't pass given your current implementation?

Reply:
Sure thing, I have just added the relevant change, thanks for noticing.